src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const copyOfState = { ...state };
+
+  for (const action of actions) {
+    const {
+      type,
+      extraData,
+      keysToRemove,
+    } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        Object.assign(copyOfState, extraData);
+        arrayOfStates.push({ ...copyOfState });
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of keysToRemove) {
+          delete copyOfState[key];
+        }
+        arrayOfStates.push({ ...copyOfState });
+        break;
+      }
+
+      case 'clear': {
+        for (const key in copyOfState) {
+          delete copyOfState[key];
+        }
+        arrayOfStates.push({ ...copyOfState });
```

Comment:
You do the same in all cases. Just push it after switching all cases.
Also, don't forget about the default case for error handling